src/components/game-modal.jsx

```diff
@@ -111,28 +111,32 @@ const GameModal = React.createClass({
   },
   asyncStartEngine(gistId, gameData) {
     // Load the game engine from a separate bundle
+    const that = this;
     /*eslint-disable no-undef */
     require.ensure([], (require) => {
       const GameEngine = require('puzzle-script');
-      /*eslint-enable no-undef */
-      const canvasNode = this.refs.gameCanvas.getDOMNode();
-      const {isPlaying} = this.state;
-      if (isPlaying) {
-        GameEngine.stop(canvasNode);
-      }
-      GameEngine.setOnWinGame(() => {
-        // When the user wins a game remember that in localStorage.
-        // That way, the next game becomes available
-        this.addGamesWon(gistId);
-        /*eslint-disable no-alert */
-        alert('Congrats! You unlocked another game!\n\nThe next one is not ready yet but check back in a day or 2... it will be');
-        /*eslint-enable no-alert */
-      });
-      GameEngine.useDefaultSaveAndLoad('game-' + gistId);
-      GameEngine.start(canvasNode, gameData);
-      this.setLastPlayed(gistId);
-      this.setState({isPlaying: true});
+      that._startEngine(GameEngine, gistId, gameData);
+    });
+  },
+  _startEngine(GameEngine, gistId, gameData) {
+    /*eslint-enable no-undef */
+    const canvasNode = this.refs.gameCanvas;
+    const {isPlaying} = this.state;
+    if (isPlaying) {
+      GameEngine.stop(canvasNode);
+    }
+    GameEngine.setOnWinGame(() => {
+      // When the user wins a game remember that in localStorage.
+      // That way, the next game becomes available
+      this.addGamesWon(gistId);
+      /*eslint-disable no-alert */
+      alert('Congrats! You unlocked another game!\n\nThe next one is not ready yet but check back in a day or 2... it will be');
+      /*eslint-enable no-alert */
     });
+    GameEngine.useDefaultSaveAndLoad('game-' + gistId);
+    GameEngine.start(canvasNode, gameData);
+    this.setLastPlayed(gistId);
+    this.setState({isPlaying: true});
   },
   onHide() {
```

Comment:
don't need to change this as well? if the issue was scoping, it's probably here too